src/telescope_baseline/mapping/aperture.py

```diff
@@ -0,0 +1,293 @@
+    """compute a single sqaure convex
+    Args: 
+        center: theta, phi of the center
+        PA: position angle of the detector (north up)
+        width: detector width in radian
+    Returns:
+        detecter convex (ang)
+    
+    """
+    anglist=[-3.0*np.pi/4.0,-np.pi/4.0,np.pi/4.0,3.0*np.pi/4.0]
+    scale=1.0/np.sqrt(2.0)
+    return basic_convex(center,PA,width,anglist,scale)
+
+def ang_detector_unit(direction,scale,center,PA,width):
+    """compute a relative ang position in the unit of the detectir size from the original position center/PA/width
+    Args: 
+        direction: TBLR =top,bottom,left,right
+        scale: scale in the detector one side unit
+        center: theta, phi of the center
+        PA: position angle of the detector (north up)
+        width: detector width in radian
+    Returns:
+        ang (theta, phi)
+    
+    """
+
+    anglist=[-np.pi/2.0,0.0,np.pi/2.0,np.pi]
+    dic={"R":0,"B":1,"L":2,"T":3}
+    bc=basic_convex(center,PA,width,anglist,2*scale)
+    i=dic[direction]
+    return bc[:,i]
+    
+def lb_detector_unit(direction,scale,l_center,b_center, PA_deg, width_mm=22.4, EFL_mm=4370.0):
+    """compute a relative l,b position in the unit of the detectir size from the original position center/PA/width
+    Args: 
+        direction: TBLR =top,bottom,left,right
+        scale: scale in the detector one side unit
+        l_center: center of galactic coordinate, l (deg)
+        b_center: center of galactic coordinate, b (deg)
+        PA_deg: position angle in deg
+        width_mm: the separation of detector chips
+        EFL_mm: effective focal length in the unit of mm
+    Returns:
+        l,b
+    
+    """
+    PA=PA_deg/180.0*np.pi
+
+    return ang2lb(ang_detector_unit(direction,scale,lb2ang(l_center,b_center),PA,width_mm/EFL_mm))
+    
+def Roty(theta):
+    return np.array([[np.cos(theta),0.0,np.sin(theta)],[0.0,1.0,0.0],[-np.sin(theta),0.0,np.cos(theta)]])
+
+def Rotz(theta):
+    return np.array([[np.cos(theta),-np.sin(theta),0.0],[np.sin(theta),np.cos(theta),0.0],[0.0,0.0,1.0]])
+
+def vec2ring(x):
+    """convert a vector to a ring
+    Args:
+        vertex list (vertex, coodinate)
+
+    Returns:
+        ring (edge, left-right, coordinate)
+
+    """
+    Nvertex, Ncoordinate=np.shape(x)
+    if Nvertex%2 == 1:
+        print("Nvertex needs to be even")
+    rs=(int(Nvertex/2),2,Ncoordinate)
+    x1=x.reshape(rs) 
+    x0=np.roll(x,1,axis=0).reshape(rs)
+    return np.hstack([x0,x1]).reshape((Nvertex,2,Ncoordinate))
+
+def cos_angle_from_normal_vectorAB(a,b,x):
+    """cos angle between the normal vector defined by a, b and x
+    Args:
+       a: 3D vetcor of A, (3,) or (N, 3)
+       b: 3D vector of B, (3,) or (N, 3)
+       x: vectors whose cos angle from normal vector of the ABO plane to be computed 3 or (M, 3)
+     
+    Returns:
+       cos_angles, float or (N,M)
+    """
+    return (x@np.cross(a,b).T).T
+
+
+def convex_on_sphere(angv,angw):
```

Comment:
The function name does not represent its functionality.
Reconsider the function name.

Reply:
right. I changed

Reply:
thx!